taichi/unified_allocator.cpp

```diff
@@ -35,7 +35,7 @@ UnifiedAllocator::UnifiedAllocator(std::size_t size, Arch arch)
     // Assuming ARM devices have shared CPU/GPU memory and do no support
     // memAdvise; CUDA on Windows has limited support for unified memory
     check_cuda_error_as_warning(
-        cudaMemAdvise(_cuda_data, size, cudaMemAdviseSetPreferredLocation, 0));
+        cudaMemAdvise(_cuda_data, size, cudaMemAdviseSetPreferredLocation, cudaCpuDeviceId));
```

Comment:
How do you think?

Reply:
Here we actually want the memory pages to stick to GPU memory as much as possible. `cudaCpuDeviceId=-1` will make them stick to CPU and possibly slow down GPU kernels.

I would stay with `0` here :-)

Reply:
Btw, let's focus on one problem in a single PR! This will make everyone's lift easier :-)

Reply:
Yeah, of course, I will change this back after everything done, or my local env won't allow me to test at all :)